_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+> to explain where and why reality has diverged from the plan.
+> Doing this extends the useful life of that document,
+> but more importantly,
+> it's a chance to reflect on what the code is teaching us
+> and share that with other people
+> so that our next design(s) can be better.
+
+### Weekly Friday Engineering Demos
+
+One of our more experienced team members instituted Friday afternoon demo sessions last year,
+and we have all become believers.
+Some demos show off a new feature;
+others walk the audience through a bug that has been fixed
+or introduce the team to useful third-party tools.
+Demos runs anywhere from a few minutes to half an hour,
+with most being near the low end of that range.
+It's a fun way to wrap up the week and remind ourselves that we're actually making progress.
+
+-   Summary: Weekly sync for team to share ongoing work and discoveries
+-   Key Questions:
+    -   It does *what*??
+-   Frequency: weekly
+-   Responsible: volunteer engineer
+-   Contributors: entire engineering team
+-   Products:
+    -   Team spirit
+    -   Lateral knowledge transfer
+
+### User Support
+
+As mentioned earlier,
+we now take week-long turns monitoring the `#eng-support` channel
+and responding to users' problems.
+In some cases (e.g., when it's my turn to do support)
+the response usually takes the form of,
+"OK, let me pull in someone who actually understands what's going on here,"
+but whether the on-call engineer handles the problem themself or hands it off,
+it gives us all regular reminders of how our software is being used
+and what state it is actually in.
+
+-   Summary: Ongoing support of issues and requests
+-   Frequency: ongoing
+-   Responsible: designated engineer (rotation)
+-   Contributors: team members as needed
+-   Products:
+    -   Happier users
+    -   Issues in the issue tracker
+
+> The "issues" product deserves a bit more explanation.
+> The on-call engineer creates a new issue in our issue tracker (we use [Linear][linear])
+> for *every* incident
+> so that there is a single point of reference for the conversation.
+> Doing this saves people from having to hunt through GitHub, a wiki,
+> and various Slack channels and/or Google docs
+> when there's a hand-off
+> or the issue occurs again.
+
+### Fixit Sprints
+
+We work in two-weeks sprints,
+and for the last year and a half,
+we've dedicated one sprint each quarter to paying down technical debt.
+If we need to clean up the data access layer in an application
+or migrate to a backward-incompatible version of a key library,
+we put that work in "our" sprint
+so that it won't disrupt work on user-facing changes.
+The benefits are hard to quantify but very real,
+and making this a regular activity makes the time easier to defend.
+
+-   Summary: 1-2 week dedicated period to address technical debt
+-   Frequency: quarterly
+-   Responsible: VP Engineering
+-   Contributors: entire engineering team
+-   Products:
+    -   Less technical debt
```

Comment:
I'd add Team spirit and happier/more productive engineers?

Reply:
added